docs/ocis/development/extensions.md

```diff
@@ -7,227 +7,70 @@ geekdocEditPath: edit/master/docs/ocis/development
 
-go back to the ocis-extension-workshop folder
-
-```console
-cd ..
-```
-
-Following https://github.com/owncloud/ocis-hello
-
-```
-git clone https://github.com/owncloud/ocis-hello.git
-cd ocis-hello
-
-yarn install
-# this actually creates the assets
-yarn build
-
-# this will compile the assets into the binary
-make generate build
-```
-
-Two options:
-1. run only the necessary services from ocis and ocis-hello independently
-2. compile ocis with the updated ocis-hello
-
-### Option 1:
-get a list of ocis services:
-
-```console
-ps ax | grep ocis
-```
-
-Try to kill `ocis hello`
-
-Remember: For now, killing a service will cause ocis to restart it. This is subject to change.
-
-In order to be able to manage the processes ourselves we need to start them independently:
-
-`bin/ocis server` starts the same services as:
-
-```
-bin/ocis micro &
-bin/ocis web &
-bin/ocis hello &
-bin/ocis reva &
-```
-
-Now we can kill the `ocis hello` and use our custom built ocis-hello binary:
+### ownCloud Web
 
-```console
-cd ../ocis-hello
-bin/ocis-hello server
-```
+If your extension is not just doing something in the background, you will need a UI in order to allow the user to interact with your extension. You could just provide your own web frontend for that purpose but for a better user experience you need to integrate in the web frontend of oCIS, [ownCloud Web](https://github.com/owncloud/web).
 
-## Hacking ownCloud Web (and ocis-web)
+ownCloud Web allows you to write an extension to itself and offers therefore a seamless user experience. The user will be able to use the application switcher to switch between the files view, settings and available extensions. Furthermore it is also possible to register your extension for different file actions. You could offer your extension to the user for creating and editing office text documents. The user will then be able to create or open a file with your application directly from the files view. How to provide create an extension for ownCloud Web can be seen best in [the Hello extension](https://github.com/owncloud/ocis-hello/blob/master/ui/app.js), whereas plain file handling without any web frontend is available in the [WOPI server extension](https://github.com/owncloud/ocis-wopiserver/blob/master/ui/app.js).
 
-Following https://github.com/owncloud/web we are going to build the current ownCloud Web
+In order to ownCloud Web pick up your extension, you need to activate it in the configuration like seen in the [Hello extension](https://owncloud.dev/extensions/ocis_hello/running/#configure-and-start-ocis). As of now this is a manual task but later on this will be done automatically by service discovery.
 
-```
-git clone https://github.com/owncloud/web.git
-cd web
+For better visual integration ownCloud Web itself uses a external design library, the [ownCloud design system](https://github.com/owncloud/owncloud-design-system). This should also be used by your extension.
```

Comment:
```suggestion
For a consistent look and feel, ownCloud Web uses a external design library, the [ownCloud design system](https://github.com/owncloud/owncloud-design-system). Since its classes and components are available through the wrapping `web runtime`, we highly recommend you to leverage it in your extension as well.
```